mscr-ui/src/modules/crosswalk-editor/tabs/edit-crosswalk/schema-info/index.tsx

```diff
@@ -0,0 +1,303 @@
+        item.name.toLowerCase().includes(nameToFind.toLowerCase())
+      ) {
+        results.nodeIds.push(item.id);
+        if (item.children && item.children.length > 0) {
+          item.children.forEach((child) => {
+            results.childNodeIds.push(child.id);
+          });
+        }
+      }
+      if (item.children && item.children.length > 0) {
+        return doFiltering(item.children, nameToFind, results);
+      }
+    });
+    return results;
+  }
+
+  function getElementPathsFromTree(
+    treeData: RenderTree[],
+    nodeIds: string[],
+    results: string[],
+  ) {
+    treeData.forEach((item) => {
+      if (nodeIds.includes(item.id)) {
+        if (item.elementPath != null) {
+          results.push(item.elementPath);
+        }
+      }
+      if (item.children && item.children.length > 0) {
+        return getElementPathsFromTree(item.children, nodeIds, results);
+      }
+    });
+    return results;
+  }
+
+  // Used by tree select and filtering
+  function getAllNodeIdsOnPathToLeaf(nodeIds: string[]) {
+    const elementPaths = getElementPathsFromTree(treeData, nodeIds, []);
+    const nodesToSelect: Set<string> = new Set();
+
+    elementPaths.forEach((path) => {
+      const nodes = path.split('.');
+      nodes.forEach((node) => {
+        nodesToSelect.add(node);
+      });
+    });
+    return Array.from(nodesToSelect);
+  }
+
+  const handleExpandClick = () => {
+    const allTreeNodes: string[] = [];
+    treeData.forEach(({ children, id }) => {
+      if (children && children?.length > 0) {
+        allTreeNodes.push(id.toString());
+      }
+    });
+    setTreeExpanded((oldExpanded) => {
+      return oldExpanded.length === 0 ? allTreeNodes : [];
+    });
+  };
+
+  function expandAndSelectNodes(nodeIds: string[]) {
+    if (nodeIds.length > 0) {
+      const nodeIdsToExpand = getAllNodeIdsOnPathToLeaf(nodeIds);
+      setTreeExpanded(nodeIdsToExpand);
+      setTreeSelections(nodeIds);
+    }
+  }
+
+  function searchFromTree(input: any) {
+    clearTreeSearch();
+    const hits = { nodeIds: [], childNodeIds: [] };
+    doFiltering(treeData, input.toString(), hits);
+    expandAndSelectNodes(hits.nodeIds);
+  }
+
+  function handleTreeClick(
+    event: React.SyntheticEvent | undefined,
+    nodeIds: string[],
+  ) {
+    setTreeSelections(nodeIds);
+  }
+
+  function handleTreeToggle(
+    event: React.SyntheticEvent | undefined,
+    nodeIds: string[],
+  ) {
+    setTreeExpanded(nodeIds);
+  }
+
+  const selectFromTreeById = (nodeId: string) => {
+    const nodeIds = [];
+    nodeIds.push(nodeId);
+    expandAndSelectNodes(nodeIds);
+  };
+
+  const performCallbackFromTreeAction = (
+    action: any,
+    event: any,
+    nodeIds: any,
+  ) => {
+    if (action === 'handleSelect') {
+      handleTreeClick(event, nodeIds);
+    } else if (action === 'treeToggle') {
+      handleTreeToggle(event, nodeIds);
+    }
+  };
+
+  const performNodeInfoAction = (nodeId: any, isSourceTree: boolean) => {
```

Comment:
Is this for some future use? NodeInfo doesn't seem to be using the performNodeInfoAction as of now. If it's not going to be in use, both this and selectFromTreeById don't have a use.

Reply:
If multiple nodes are selected from tree, there could be a possibility to select the node to be mapped from the node drop down. But this is a stub and some more discussion about details is needed.